docs/rules/no-attributes-during-construction.md

```diff
@@ -0,0 +1,50 @@
+# Disallow setting attributes during construction (no-attributes-during-construction)
+
+The `LightningElement` base class extended by LWC component classes defines several properties that, when set,
+renders attributes on its custom element. This behavior mimics the native browser behavior.
+
+By [specification](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-element-conformance), custom
+element constructors must not cause the custom element to gain attributes. This rule prevents set operations in the
+constructor method that violate this restriction.
+
+This rule only knows about `LightningElement` properties that implement this behavior (e.g., `hidden`, `id`, `role`,
+`tabIndex`, `title`, etc) and will not detect custom implementations that may set attributes during construction time.
```

Comment:
Should we add a "Caveats" (or similar) section like in [this rule](https://github.com/salesforce/eslint-plugin-lwc/blob/master/docs/rules/no-unexpected-wire-adapter-usages.md#caveats), mentioning that this rule won't detect violations on component classes that do not directly inherit from `LightningElement`?

```js
import { LightningElement } from 'lwc';
class Foo extends LightningElement {};
class Test extends Foo {
    constructor() {
        this.tabIndex = '0';  // valid, though is incorrect.
    }
}
```